File: EXAM/pasquel/u3/PasquelFinalExam/model/quick_sort.py
def quick_sort(new_array):
    if len(new_array) <= 1:
        return new_array
    pivot = new_array[len(new_array) // 2]
    left = [x for x in new_array if x < pivot]
    middle = [x for x in new_array if x == pivot]
    right = [x for x in new_array if x > pivot]
    return quick_sort(left) + middle + quick_sort(right)

File: EXAM/maza/u3/ExamU3/model/QuickSort.js
export class QuickSort {
    static sort(arr) {
        if (arr.length <= 1) return arr;
        let pivot = arr[arr.length - 1];
        let left = arr.filter((el, i) => el <= pivot && i !== arr.length - 1);
        let right = arr.filter(el => el > pivot);
        return [...QuickSort.sort(left), pivot, ...QuickSort.sort(right)];
    }
}
